File: examples/python/guides/integrations/embedding_openai.py
# Third-party integration example - requires: pip install openai
# See: /guides/rag-and-indexing

from openai import OpenAI

client = OpenAI()


# > OpenAI embedding usage
def embed(text: str) -> list[float]:
    r = client.embeddings.create(model="text-embedding-3-small", input=text)
    return r.data[0].embedding

File: examples/go/guides/integrations/embedding_openai.go
// Third-party integration - requires: go get github.com/sashabaranov/go-openai
// See: /guides/rag-and-indexing

package integrations

import (
	"context"
	"os"

	"github.com/sashabaranov/go-openai"
)

// > OpenAI embedding usage
func Embed(ctx context.Context, text string) ([]float32, error) {
	client := openai.NewClient(os.Getenv("OPENAI_API_KEY"))
	resp, err := client.CreateEmbeddings(ctx, openai.EmbeddingRequest{
		Model: openai.AdaEmbeddingV2,
		Input: text,
	})
	if err != nil {
		return nil, err
	}
	return resp.Data[0].Embedding, nil
}

